File: C#/FTPResponseExporter/FTPResponseExporter/service/Authentication.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ServiceModel;
using FTPResponseExporter.AuthService;

namespace ResponseExporter.service
{
    public class Authentication
    {
        private static readonly String WSDL_LOCATIONS = "Services/AuthenticationService.svc/soap";
        private AuthenticationServiceClient client;
      

        public Authentication(Uri APIUrl)
        {
            
            EndpointAddress endpoint = new EndpointAddress(APIUrl.ToString() + WSDL_LOCATIONS);
            if ("https".Equals(APIUrl.Scheme))
                client = new AuthenticationServiceClient("BasicHttpBinding_IAuthenticationService11", endpoint);
            else if ("http".Equals(APIUrl.Scheme))
                client = new AuthenticationServiceClient("BasicHttpBinding_IAuthenticationService", endpoint);
         
        }


        public String getAuthenticationToken(String userName, String password)
        {
            Console.WriteLine("Authenticating");
            ServiceOperationResultOfstring loginServiceResult = client.Login(userName, password);
            if (!loginServiceResult.CallSuccess)
            { Console.WriteLine("Authentication Failed"); }
            else
            { Console.Write("Authentication Success:"); }
         ;
            
            return loginServiceResult.ResultData;
            

        }
     
    }
}

File: Java/FTPResponseExporter/src/main/java/com/checkbox/ftpresponseexporter/service/Authentication.java
package com.checkbox.ftpresponseexporter.service;

import com.checkbox.ftpresponseexporter.Main;
import com.checkbox.ftpresponseexporter.authentication.AuthenticationService;
import com.checkbox.ftpresponseexporter.authentication.IAuthenticationService;
import com.checkbox.ftpresponseexporter.authentication.ServiceOperationResultOfstring;
import com.checkbox.ftpresponseexporter.exception.LoginFailedException;

import java.net.MalformedURLException;
import java.net.URL;



public class Authentication {

    private static final String WSDL_LOCATION = "/Services/AuthenticationService.svc?wsdl";
    private AuthenticationService service;
    private IAuthenticationService client;

    public Authentication(URL APIUrl) throws MalformedURLException {
        service = new AuthenticationService(new URL(APIUrl.toString() + WSDL_LOCATION));
        if ("https".equals(APIUrl.getProtocol())) {
            client = service.getBasicHttpBindingIAuthenticationService1();
        } else {
            client = service.getBasicHttpBindingIAuthenticationService();
        }
    }

    public String getAuthenticationToken(String userName, String password) throws LoginFailedException {
        Main.LOGGER.info(userName + " authentication...");
        ServiceOperationResultOfstring loginServiceResult = client.login(userName, password);
        if (!loginServiceResult.isCallSuccess()) {
            throw new LoginFailedException(loginServiceResult.getFailureMessage().getValue());
        }
        return loginServiceResult.getResultData().getValue();
    }
}
